# SPDX-FileCopyrightText: The AdaptiveCpp Contributors
# SPDX-FileCopyrightText: 2026 MistVVK and the XeStrata contributors
# SPDX-License-Identifier: BSD-2-Clause
#
# Copyright The AdaptiveCpp Contributors. All rights reserved.
# Modified 2026 by MistVVK and the XeStrata contributors:
# retain the supplied async_handler in sycl::event.
# Original event.hpp copyright and BSD-2-Clause notice remain unchanged.
# Full license: ../LICENSE
# Applies only to AdaptiveCpp 0ea863abe1af4422593583f91e5608ef75044a16.
# This does not change queue ordering, DAG scheduling, cancellation or cleanup.

diff --git a/include/hipSYCL/sycl/event.hpp b/include/hipSYCL/sycl/event.hpp
--- a/include/hipSYCL/sycl/event.hpp
+++ b/include/hipSYCL/sycl/event.hpp
@@ -44,4 +44,4 @@ public:
       const rt::dag_node_ptr &evt,
       async_handler handler =
           [](exception_list e) { glue::default_async_handler(e); })
-      : _node{evt} {}
+      : _node{evt}, _handler{handler} {}
